tests/e2e/pageobjects/ide/ProjectTree.ts

```diff
@@ -265,9 +265,10 @@ export class ProjectTree {
         const rootSubitemLocator: By = By.css(this.getTreeItemCssLocator(`${projectName}/${rootSubItem}`));
 
         for (let i = 0; i < attempts; i++) {
-            const isProjectFolderVisible = await this.driverHelper.waitVisibilityBoolean(rootItemLocator, attempts, visibilityItemPolling);
+            const isProjectFolderVisible = await this.driverHelper.waitVisibilityBoolean(rootItemLocator, attempts, visibilityItemPolling); // todo: the number of attempts in the argument of the method is meaningless, because the number of attempts is being squared here
```

Comment:
Please I need input on this issue.
The logic here is as follows
```
for ( i=0; i<attempts; i++) {
...
this.driverHelper.waitVisibilityBoolean(... , attempts, visibilityItemPolling)
...
}
```
The resulting number of attempts is then as follows
`num_attempts = attempts * attempts`
because of the use in a for loop and then it being passed as an argument into a function

Reply:
What about having different variables for those cycles? There can be `attempts` used in `for` cycle, but in the method, there can be specific variable `projectItemAttempts`. That way it can be clear what each `attempts` variable is exactly doing. WDYT?

Reply:
I'm up for modifying the method, but that would add another argument to the method itself and would require changes in tests as well. If that's not an issue I would be happy to implement the change

Reply:
If you don't want you don't have to change the method description. You can just add a new variable to TestConstats and then reference it directly in the method. E.g. something like that:
```
const isProjectFolderVisible = await this.driverHelper.waitVisibilityBoolean(rootItemLocator, TestConstants.TS_SELENIUM_PROJECT_ITEM_VISIBILITY_ATTEMPTS, visibilityItemPolling);
```

Reply:
sounds good, but just to keep the logic consistent with the arguments provided to the method, I would just statically assign `waitVisibilityBoolean` `1` attempt

Reply:
I don't mind that approach if it will not mean that the page will be refreshed too often. I think it requires testing and seeing how the test behaves. :) maybe @Ohrimenko1988 will have some more ideas here

Reply:
The fix seems to be working fine. I will squash the commits together, remove WIP status and resolve this conversation